bin/watch.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Script to watch for changes to actions and upload them to the recipe
+ * server.
+ */
+import 'babel-polyfill';
+
+import path from 'path';
+
+import chokidar from 'chokidar';
+import {argv} from 'yargs';
+
+import {NormandyApi} from '../lib/api';
+import {Action} from '../lib/models';
+import {loadConfig, localPath} from '../lib/utils';
+
+
+let config = loadConfig(argv.env);
+if (config === null) {
+    process.exit();
+}
+
+let api = new NormandyApi(config.api_token, config.normandy_url, config.verify);
+let watcher = chokidar.watch(localPath('actions'));
+watcher.on('change', async function(changedPath) {
+    let actionName = path.relative(localPath('actions'), changedPath).split(path.sep)[0];
+    let action = new Action(actionName);
+
+    if (action.isValid) {
```

Comment:
Maybe. The path logic is a bit fiddly, so the `isValid` check also acts as a validation that what changed was an action at all vs some other random file. The main case that would slip out from here is an action missing an entrypoint or a schema, which IMO is fairly obvious and not something we need to be loud about.

I'm on the fence leaning towards no but am happy to add such a warning in if you'd like.